java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -765,21 +799,39 @@ private void initAndResizeOffsetBuffer(int currentIndex) {
       }
     }
 
-    private void initValidBuffer() {
+    private void initAndResizeValidBuffer() {
       if (this.valid == null) {
         allocateBitmaskAndSetDefaultValues();
+      } else {
+        long bitmaskSize = ColumnVector.getNativeValidPointerSize((int) rows);
```

Comment:
This is still reallocating memory every X entries.  Don't we want this to grow at the same rate that we grow the data buffer?

Reply:
yeah, I will hopefully consolidate this based on my and your comment from earlier that we can increment and grow buffers all at once.

Reply:
>This is still reallocating memory every X entries.
>
Only if valid length < new bitmaskSize. since now we have validity set based on estimated rows I dont think we will hit this as often, except for the native call.

Reply:
I think now that this code sits with the data buffer , it should be ok? If this is not coupled with data buffer as much as you would like, let me know.